animation_nodes/algorithms/mesh/triangulate_mesh.pyx

```diff
@@ -0,0 +1,316 @@
+        for j in range(polyLength):
+            angle = calculateAngleWithSign(verticesData, j)
+            verticesData[j].angle = angle
+            if angle >= 0.0 and not pointsInTriangle(polyLength, verticesData, j):
+                verticesData[j].isEar = True
+            else:
+                verticesData[j].isEar = False
+
+        # Calculate triangle polygon indices.
+        for j in range(polyLength - 2):
+            if polyLength - 2 - j == 1:
+                newPolyStarts[triangleIndex] = polyIndex
+                newPolyLengths[triangleIndex] = 3
+                triangleIndex += 1
+
+                index = 0
+                for k in range(polyLength):
+                    if verticesData[k].isEar:
+                        newIndices[polyIndex + index] = verticesData[k].index
+                        index += 1
+                        if index == 3: break
+                polyIndex += 3
+                break
+
+            # Removing an ear which has smallest inner-angle.
+            earIndex = findEarHasMinAngle(polyLength, verticesData)
+
+            newPolyStarts[triangleIndex] = polyIndex
+            newPolyLengths[triangleIndex] = 3
+            triangleIndex += 1
+
+            newIndices[polyIndex] = verticesData[verticesData[earIndex].previous].index
+            newIndices[polyIndex + 1] = verticesData[earIndex].index
+            newIndices[polyIndex + 2] = verticesData[verticesData[earIndex].next].index
+            polyIndex += 3
+
+            removeEarVertex(polyLength, verticesData, earIndex)
+
+        PyMem_Free(verticesData)
+
+    return newPolygons
+
+# Remove ear-vertex and update neighbor-vertices.
+cdef void removeEarVertex(Py_ssize_t polyLength, Vertex* verticesData, Py_ssize_t earIndex):
+    cdef Py_ssize_t previousIndex, nextIndex
+    previousIndex = verticesData[earIndex].previous
+    nextIndex = verticesData[earIndex].next
+
+    verticesData[earIndex].isEar = False
+    verticesData[nextIndex].previous = previousIndex
+    verticesData[previousIndex].next = nextIndex
+
+    cdef float angle
+    angle = calculateAngleWithSign(verticesData, previousIndex)
+    verticesData[previousIndex].angle = angle
+    if angle >= 0.0 and not pointsInTriangle(polyLength, verticesData, previousIndex):
+        verticesData[previousIndex].isEar = True
+    else:
+        verticesData[previousIndex].isEar = False
+
+    angle = calculateAngleWithSign(verticesData, nextIndex)
+    verticesData[nextIndex].angle = angle
+    if angle >= 0.0 and not pointsInTriangle(polyLength, verticesData, nextIndex):
+        verticesData[nextIndex].isEar = True
+    else:
+        verticesData[nextIndex].isEar = False
+
+cdef struct Vertex:
+    Py_ssize_t index
+    Vector3 location
+    float angle
+    bint isEar
+    Py_ssize_t previous
+    Py_ssize_t next
+
+# Make sure normal (+z) is correct.
+@cython.cdivision(True)
+cdef bint polyCounterClockwise(Vertex* verticesData):
```

Comment:
For functions that return bint, name it as `is*`, in this case, `isPolyCounterClockwise`.

Reply:
Okay. This function is changed to `makeSurePolygonIsCounterClockwise`.